frontend/src/api/Socket.ts

```diff
@@ -3,14 +3,52 @@ import Stomp from 'stompjs';
 
 let stompClient:any = null;
 
-export const connect = (endpoint:string) => {
-  const socket:any = new SockJS(endpoint);
-  stompClient = Stomp.over(socket);
-  stompClient.connect();
+type User = {
+  nickname: string;
+}
+
+// Variable to hold the current connected state.
+let connected = false;
+
+// Create constants for the subscription and send message URLs.
+const SUBSCRIBE_URL:string = '/api/v1/socket/subscribe-user';
+const SEND_USER_URL:string = '/api/v1/socket/user';
+
+export const sendUser = (nickname:string) => {
+  if (connected) {
+    const nicknameInput:string = (nickname !== '') ? nickname : 'Anonymous';
+    stompClient.send(SEND_USER_URL, {}, nicknameInput);
+  } else {
+    console.error('You must be connected to a socket before sending user information.');
+  }
+};
+
+export const connect = (endpoint:string, nickname:string) => {
+  if (!connected) {
+    // Connect to given endpoint, subscribe to future messages, and send user message.
+    const socket:any = new SockJS(endpoint);
+    stompClient = Stomp.over(socket);
+    stompClient.connect({}, () => {
+      stompClient.subscribe(SUBSCRIBE_URL, (user:any) => {
+        const userObject:User = JSON.parse(user.body);
+        console.log(`Welcome ${userObject.nickname} to the page!`);
+      });
+      sendUser(nickname);
+    });
+    // Reassign connected variable.
+    connected = true;
+  } else {
+    console.error('You are already connected to a socket.');
+  }
 };
 
 export const disconnect = () => {
-  if (stompClient !== null) {
+  if (connected) {
     stompClient.disconnect();
+    stompClient = null;
+    // Reassign connected variable.
+    connected = false;
+  } else {
+    console.error('You cannot disconnect because you are not connected to any socket.');
   }
 };
```

Comment:
nit: Remove or replace `any` tags, if possible

Reply:
I didn't realize how to do that before -- I was able to find the relevant types for looking at the files from the `import` statements.

I also added this to the rules in `.eslintrc.json`:
```
  "no-unused-vars": "off",
  "@typescript-eslint/no-unused-vars": [
    "error"
  ],
```
This (found at [this Stack Overflow post](https://stackoverflow.com/a/61555310/7517518)) change fixes the false positives for the `no-unused-vars` warnings.